waves_gateway/service/support_log_handler.py

```diff
@@ -0,0 +1,110 @@
+"""
+SupportLogHandler
+"""
+import uuid
+from logging import Handler
+
+from waves_gateway.common import Injectable
+from waves_gateway.storage.log_storage_service import LogStorageService
+
+
+@Injectable(deps=[LogStorageService], provides=Handler)
+class SupportLogHandler(Handler):
+    """
+    SupportLogHandler persists logs for support
+    """
+
+    def __init__(self, log_storage_service: LogStorageService) -> None:
+        Handler.__init__(self)
+        self._log_storage_service = log_storage_service
+
+    # will be called from the log handler by logger.info / .error / .debug
+    def emit(self, record):
+        log_entry = self.format(record)
+        datetime = log_entry[1:20]
+        level = record.levelname
+        affected_component = log_entry.split(" ")[3]
```

Comment:
This depends on the formatter being used which is currently being set in the concrete Gateway implementation. So, you cannot expect the string to be formatted like this.
For this to work, the formatting must be forced and moved into the framework.

Reply:
This would be a breaking change.